tensor2tensor/data_generators/ptb.py

```diff
@@ -0,0 +1,158 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Data generators for PTB data-sets."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import sys
+import tarfile
+import collections
+
+# Dependency imports
+
+from tensor2tensor.data_generators import generator_utils
+from tensor2tensor.data_generators import text_encoder
+
+import tensorflow as tf
+
+
+EOS = text_encoder.EOS
+PTB_URL = "http://www.fit.vutbr.cz/~imikolov/rnnlm/simple-examples.tgz"
+
+def _read_words(filename):
+  """Reads words from a file.
+      It returns a list of words without '\n'
+      Originally from: 
+      https://github.com/tensorflow/models/blob/master/tutorials/rnn/ptb/reader.py
+  """
+  with tf.gfile.GFile(filename, "r") as f:
+    if sys.version_info[0] >= 3:
+      return f.read().replace("\n", " ").split()
+    else:
+      return f.read().decode("utf-8").replace("\n", " ").split()
+       
+       
+
+def _build_vocab(filename, vocab_path, vocab_size, exclude=[]):
```

Comment:
Please use `generator_utils.get_or_generate_vocab` instead of this private function. This allows to later run models in multi-problem mode (only possible with matching vocabs).